pkg/operator/controllers/pullsecret/pullsecret_controller.go

```diff
@@ -104,10 +109,11 @@ func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.
 	// - list of Red Hat pull-secret keys in status.
 	instance.Status.RedHatKeysPresent, err = r.parseRedHatKeys(userSecret)
 	if err != nil {
+		r.Log.Error(err)
 		return reconcile.Result{}, err
 	}
 
-	err = r.client.Update(ctx, instance)
+	err = r.Client.Update(ctx, instance)
 	return reconcile.Result{}, err
```

Comment:
We should validate if err is nil or not here, and:
- if err is nil, call the ClearConditions function to reset the controller back to Available: True, Progressing: False, Degraded: False
- if err is not nil, call SetDegraded to set the degraded status.

Reply:
This comment has been addressed